samples/bluetooth/bap_broadcast_sink/src/main.c

```diff
@@ -732,89 +676,93 @@ static bool find_valid_bis_in_subgroup_bis(const struct bt_bap_base_subgroup *su
  *    configuration overwrites the subgroup values
  *
  * This function returns `true` if the subgroup does not support the channels in
- * CONFIG_TARGET_BROADCAST_CHANNEL which makes it iterate over the next subgroup, and returns
- * `false` if this subgroup satisfies our CONFIG_TARGET_BROADCAST_CHANNEL.
+ * CONFIG_TARGET_BROADCAST_CHANNEL which makes it iterate over the next subgroup, and
+ * returns `false` if this subgroup satisfies our CONFIG_TARGET_BROADCAST_CHANNEL.
  */
-static bool find_valid_bis_in_subgroup_cb(const struct bt_bap_base_subgroup *subgroup,
-					  void *user_data)
+static bool subgroup_get_valid_bis_indexes_cb(const struct bt_bap_base_subgroup *subgroup,
+					      void *user_data)
 {
-	enum bt_audio_location chan_allocation;
+	enum bt_audio_location subgroup_chan_allocation;
+	enum bt_audio_location chan_allocation = BT_AUDIO_LOCATION_MONO_AUDIO;
+	bool subgroup_chan_allocation_available = false;
 	struct bt_audio_codec_cfg codec_cfg;
-	uint32_t *bis_indexes = user_data;
+	struct bis_channel_allocation_data data = {
+		.cnt = 0,
+	};
+	uint32_t bis_indexes = 0;
 	int err;
 
-	/* We only want indexes from a single subgroup, so reset between each of them*/
-	*bis_indexes = 0U;
-
 	err = bt_bap_base_subgroup_codec_to_codec_cfg(subgroup, &codec_cfg);
 	if (err != 0) {
 		printk("Could not get codec configuration: %d\n", err);
 
-		return true;
+		return true; /* continue to next subgroup */
 	}
 
-	err = bt_audio_codec_cfg_get_chan_allocation(&codec_cfg, &chan_allocation, false);
-	if (err != 0) {
-		printk("Could not find subgroup channel allocation: %d - Looking in the BISes\n",
-		       err);
+	if (codec_cfg.id != BT_HCI_CODING_FORMAT_LC3) {
+		/* Only LC3 codec supported */
+		return false; /* abort */
+	}
 
-		/* Find chan alloc in BIS */
-		if (find_valid_bis_in_subgroup_bis(subgroup, bis_indexes)) {
-			/* Found BISes with correct channel allocation */
-			return false;
-		}
+	/* Get channel allocation at subgroup level */
+	err = bt_audio_codec_cfg_get_chan_allocation(&codec_cfg, &subgroup_chan_allocation, true);
+	if (err == 0) {
+		printk("Channel allocation (subgroup level) 0x%x\n", subgroup_chan_allocation);
+		subgroup_chan_allocation_available = true;
+	} else if (err == -ENODATA) {
+		printk("No channel allocation data in subgroup\n");
```

Comment:
```suggestion
```

Since we have checked for LC3 and using fallback behavior, this cannot happen

Reply:
Right. Well spotted!